client.go

```diff
@@ -0,0 +1,144 @@
+package stream_chat
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/pascaldekloe/jwt"
+)
+
+const (
+	defaultBaseURL = "https://chat-us-east-1.stream-io-api.com"
+	defaultTimeout = 6 * time.Second
+)
+
+type client struct {
+	baseURL   string
+	apiKey    string
+	apiSecret []byte
+	authToken string
+	timeout   time.Duration
+	http      *http.Client
+}
+
+func (c *client) setHeaders(r *http.Request) {
+	r.Header.Set("Content-Type", "application/json")
+	r.Header.Set("X-Stream-Client", "stream-go-client")
+	r.Header.Set("Authorization", c.authToken)
+	r.Header.Set("stream-auth-type", "jwt")
+}
+
+func (c *client) parseResponse(resp *http.Response, result interface{}) error {
+	if resp.Body != nil {
+		defer resp.Body.Close()
+	}
+
+	if resp.StatusCode >= 399 {
```

Comment:
it's not clear what do you mean

Reply:
The code as is suggests we can process, say a redirect [3xx]. Make it clear that we need an HTTP OK [2xx].

Reply:
In other  implementations of stream client there are used `res.StatusCode >= 399`, why in go it should have different behavior?

Reply:
Because it's plain wrong. The following code assumes an HTTP success [2xx] so then code it exactly like that. No biggie. Use `if status < 200 || status >= 300` if you please.